File: TaskManagerFront/task_manager_front/src/app/core/models/dtos/user-tasks/create-user-task-dto.ts
import { UserTaskPriority } from "../../enums/user-task-priority.enum";

export interface CreateUserTaskDto {
  title: string;
  description?: string | null;
  userId: number;
  additionalInfo?: AdditionalInfoDto | null;
}

export interface AdditionalInfoDto {
    priority?: UserTaskPriority | null;
    estimatedEndDate?: string | null;
    tags?: string[] | null;
}

File: TaskManagerApi/TaskManagerApi/Dtos/UserTaskDto/CreateUserTaskDto.cs
using System.ComponentModel.DataAnnotations;

namespace TaskManagerApi.Dtos.UserTaskDto
{
    public class CreateUserTaskDto
    {
        [Required(ErrorMessage = "Title is required.")]
        [StringLength(200)]
        public string Title { get; set; } = string.Empty;

        public string? Description { get; set; }

        [Required(ErrorMessage = "UserId is required.")]
        public int UserId { get; set; }

        public AdditionalInfoDto? AdditionalInfo { get; set; }
    }
}
